TST: Test pd.Grouper base floating point error (#25161)

ihsansecer opened the pull request:
- [x] closes #25161
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

ihsansecer committed:
TST: Test pd.Grouper base floating point error (#25161)

WillAyd commented on pandas/tests/groupby/test_grouping.py:
```diff
@@ -203,6 +203,18 @@ def test_grouper_creation_bug(self):
+                       "2018-11-26 16:17:43.51",
+                       periods=10, freq="1S"))
+        t0 = s.index[0]
+        base = t0.minute + (t0.second + t0.microsecond / 1e6) / 60
```
Can you replace expressions like these with literal values? I realize this is copied directly from referenced issue but writing it this way makes the test more difficult to reason about than it needs to be

ihsansecer replied:
Done! Thanks

WillAyd reviewed (changes requested):
Can you create this as a separate test? The existing one you are appending to is rather large as is

ihsansecer committed:
TST: Create a new function for testing grouper base (#25161)

ihsansecer committed:
TST: Clarify the test using literal values (#26240)

jreback commented on pandas/tests/groupby/test_grouping.py:
```diff
@@ -303,6 +303,18 @@ def test_grouper_getting_correct_binner(self):
     def test_grouper_iter(self, df):
         assert sorted(df.groupby('A').grouper) == ['bar', 'foo']
 
+    def test_grouper_base(self):
```
can you move to test pandas/tests/resample/test_timegrouper and also test in .resample (which is what this calls)

ihsansecer replied:
Changed it

ihsansecer committed:
TST: Move test to resample/test_datetime_index and rename (#25161)

ihsansecer committed:
TST: Clarify resample float base test (#25161)

Use resample() which is what Grouper calls, assert index instead of result of size()

gfyoung commented on pandas/tests/resample/test_datetime_index.py:
```diff
@@ -751,6 +751,19 @@ def test_resample_base():
+    base = 17 + 43.51 / 60
+    resampled = s.resample("3min", base=base).mean()
+    exp_index = DatetimeIndex(["2018-11-26 16:17:43.51"])
+    tm.assert_index_equal(resampled.index, exp_index)
```
FWIW, I would prefer to just check the whole `Series`, not the index.

ihsansecer replied:
I see. Will change it today

ihsansecer committed:
TST: Test if whole Series is as expected (#25161)

jreback commented:
thanks @ihsansecer
